src/StringUtility.cpp

```diff
@@ -4,5 +4,46 @@
 
 #include "StringUtility.h"
 
-#include <boost/algorithm/string/split.hpp>
 #include <boost/algorithm/string/classification.hpp>
+#include <boost/algorithm/string/split.hpp>
+#include <openssl/bio.h>
+#include <openssl/buffer.h>
+#include <openssl/evp.h>
+#include <openssl/sha.h>
+
+namespace StringUtility {
+    std::vector<std::string> split(const std::string& str, const std::string& dels) {
+        std::vector<std::string> token(0);
+        // token_compress_on means that function will reduce the empty string.
+        boost::split(token, str, boost::is_any_of(dels), boost::token_compress_on);
+        return token;
+    }
+
+    std::string sha1(const std::string& str) {
+        unsigned char output_buf[20];
+        SHA1((unsigned char*)str.c_str(), str.length(), output_buf);
```

Comment:
API Manual recommends using the EVP_* functions.
https://linux.die.net/man/3/sha

Reply:
done